automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/workspace/WorkspaceSpec.scala

```diff
@@ -128,6 +130,100 @@ class WorkspaceSpec extends FreeSpec with WebBrowserSpec with WorkspaceFixtures
+          detailPage.ui.addWorkspaceAttribute("c", "c")
+          detailPage.ui.save()
+
+          // TODO: ensure sort, for now it's default sorted by key, ascending
+          detailPage.ui.readWorkspaceTable shouldBe List(List("a", "a"), List("b", "b"), List("c", "c"))
+        }
+      }
+
+      // This table is notorious for getting out of sync
+      "should be able to correctly delete workspace attributes" - {
+        "from the top" in withWebDriver { implicit driver =>
+          withWorkspace(billingProject, "WorkspaceSpec_del_ws_attrs", attributes = Some(testAttributes)) { workspaceName =>
+            val listPage = signIn(Config.Users.harry)
+            val detailPage = listPage.openWorkspaceDetails(billingProject, workspaceName).awaitLoaded()
+
+            detailPage.ui.beginEditing()
+            detailPage.ui.deleteWorkspaceAttribute("A-key")
+            detailPage.ui.save()
+
+            detailPage.ui.readWorkspaceTable shouldBe List(List("B-key", "B value"), List("C-key", "C value"))
+          }
+        }
+
+        "from the middle" in withWebDriver { implicit driver =>
+          withWorkspace(billingProject, "WorkspaceSpec_del_ws_attrs", attributes = Some(testAttributes)) { workspaceName =>
+            val listPage = signIn(Config.Users.harry)
+            val detailPage = listPage.openWorkspaceDetails(billingProject, workspaceName).awaitLoaded()
+
+            detailPage.ui.beginEditing()
+            detailPage.ui.deleteWorkspaceAttribute("B-key")
+            detailPage.ui.save()
+
+            detailPage.ui.readWorkspaceTable shouldBe List(List("A-key", "A value"), List("C-key", "C value"))
+          }
+        }
+
+        "from the bottom" in withWebDriver { implicit driver =>
+          withWorkspace(billingProject, "WorkspaceSpec_del_ws_attrs", attributes = Some(testAttributes)) { workspaceName =>
+            val listPage = signIn(Config.Users.harry)
+            val detailPage = listPage.openWorkspaceDetails(billingProject, workspaceName).awaitLoaded()
+
+            detailPage.ui.beginEditing()
+            detailPage.ui.deleteWorkspaceAttribute("C-key")
+            detailPage.ui.save()
+
+            detailPage.ui.readWorkspaceTable shouldBe List(List("A-key", "A value"), List("B-key", "B value"))
+          }
+        }
+
+        "after adding them" in withWebDriver { implicit driver =>
+          withWorkspace(billingProject, "WorkspaceSpec_del_ws_attrs") { workspaceName =>
+            val listPage = signIn(Config.Users.harry)
+            val detailPage = listPage.openWorkspaceDetails(billingProject, workspaceName).awaitLoaded()
+
+            detailPage.ui.beginEditing()
+            detailPage.ui.addWorkspaceAttribute("a", "a")
+            detailPage.ui.addWorkspaceAttribute("b", "b")
+            detailPage.ui.addWorkspaceAttribute("c", "c")
+            detailPage.ui.save()
+
+            // TODO: better way to tell the page is ready
+            Thread sleep 1000
```

Comment:
agree with the TODO; this looks fragile, especially when we consider running this in different environments

Reply:
Yes, we should fix this. How does a person recognize that the page is ready?